status-server/src/public/js/client.js

```diff
@@ -42,15 +42,77 @@ function init() {
     }
   }
   // populate the LEDs with some initial colors
-  var colors = ['#ffff00', '#ffff00', '#00ffff', '#00ff07', '#772cb6'];
+  var colors = ['#ff00ff', '#ffff00', '#00ffff', '#00ff07', '#772cb6'];
   colors.forEach((color, idx) => {
     var node = document.getElementById("led" + idx);
     var ctx = node.getContext("2d");
     ctx.fillStyle = color;
     ctx.fillRect(0,0,300,300);
+    ctx.font = "10px Arial";
+    ctx.strokeText(idx ,node.width/2,node.height/2);
+
+
   });
 }
+function animate(ct){
+  var fetchResponse = window.fetch('http://localhost:3000/status'+ct+'.json'+Date.now()).then(
+      function onSuccess(response){
+        return response.json();
+      },
+      function onError(err){
+        console.console.warn("Error fetching"+ url, err);
+      }
+    );
+           updateValueStatus(ct);
+
+}
+
+function updateValueStatus(ct){
+
+    var  statusData = statusItems[ct];
+      console.log(statusData);
+
+    var value = statusData.value;
+    console.log(value);
+    var didChange;
+    var lastModified = statusData['last-modified'];
+    if(value == 1){
+      console.log("changed to"+0);
+      value = 0;
+      didChange=true;
+    } else if (value === 0){
+      console.log("changed to"+1);
+      value =1;
+      didChange=true;
+    }else{
+      console.log("Didn't changed");
+      didChange = false;
+    }
+    statusData.value = value;
+    statusData.lastModified = Date.now();
+    statusData.didChange = didChange;
+
 
+}
+function blinkStatus(ct){
+    var statusData = Object.keys('/status'+ct+'.json');
+    console.log(statusData);
+    if (statusData.didChange) {
+      console.log('render blink status:', '/status'+ct+'.json', statusData);
+      var nodeSelected = document.getElementsByName("iframeS1").contentWindow.getElementById("led"+ct).getContext("2d");
```

Comment:
This belongs in the renderStatus function, whose job it is to examine the statusData object and decide how to visualize any changes. Each iframe has its own client.js, and each will call its own copy of renderStatus. Its up to that function to see if didChange is true for the statusData object that represents its own state.